app/src/main/java/com/rkm/tasky/feature/error/ErrorToUiText.kt

```diff
@@ -0,0 +1,21 @@
+package com.rkm.tasky.feature.error
+
+import com.rkm.tasky.R
+import com.rkm.tasky.network.util.NetworkError
+
+fun errorToUiMessage(error: NetworkError.APIError): Int {
+    return when(error) {
+        NetworkError.APIError.NO_INTERNET -> R.string.api_error_no_internet
+        NetworkError.APIError.REQUEST_TIMEOUT ->  R.string.api_error_request_timeout
+        NetworkError.APIError.TOO_MANY_REQUESTS ->  R.string.api_error_too_many_requests
+        NetworkError.APIError.CONNECTION_TIMEOUT -> R.string.api_error_connection_timeout
+        NetworkError.APIError.PAYLOAD_TOO_LARGE ->  R.string.api_error_payload_too_large
+        NetworkError.APIError.UNAUTHORIZED ->  R.string.api_error_unauthorized
+        NetworkError.APIError.FORBIDDEN ->  R.string.api_error_forbidden
+        NetworkError.APIError.NOT_FOUND ->  R.string.api_error_not_found
+        NetworkError.APIError.SERVER_ERROR ->  R.string.api_error_server_error
+        NetworkError.APIError.SERIALIZATION ->  R.string.api_error_serialization
+        NetworkError.APIError.UNKNOWN ->  R.string.api_error_unknown
+        NetworkError.APIError.NO_SESSION_INFO ->  R.string.api_error_no_session_info
+    }
```

Comment:
Philipp has the following class to help you use string resources throughout your app, which I think you may find helpful. It allows you to pass around a regular string or a string resource without needing a context until you want to display it on the screen.

```
sealed interface UiText {
  data class DynamicString(val value: String): UiText
  class StringResource(
    @StringRes val id: Int,
    val args: Array<Any> = arrayOf()
  ): UiText

  @Composable
  fun asString(): String {
    return when(this) {
      is DynamicString -> value
      is StringResource -> stringResource(id = id, *args)
    }
  }
  
  fun asString(context: Context): String {
    return when(this) {
      is DynamicString -> value
      is StringResource -> context.getString(id, *args)
    }
  }
}
```